src/test/java/pro/sky/API/SwaggerUI/Postman/ApplicationTest.java

```diff
@@ -0,0 +1,217 @@
+import pro.sky.API.SwaggerUI.Postman.Model.Faculty;
+import pro.sky.API.SwaggerUI.Postman.Model.Student;
+import pro.sky.API.SwaggerUI.Postman.Repository.FacultyRepository;
+import pro.sky.API.SwaggerUI.Postman.Repository.StudentRepository;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ApplicationTests {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    StudentRepository studentRepository;
+    @Autowired
+    FacultyRepository facultyRepository;
+
+    @Autowired
+    private StudentController studentController;
+
+    @Autowired
+    private FacultyController facultyController;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @Test
+    void contextLoads() {
+        assertNotNull(studentController);
+        assertNotNull(facultyController);
+    }
+
+
+   private long findLastStudentId() {
+       List<Student> students = studentRepository.findAll();
+       Student lastStudent = students.stream()
+               .max(Comparator.comparing(Student::getId))
+               .orElse(null);
+       if (lastStudent == null) {
+           throw new NullPointerException("таблица со студентами пуста");
+       }
+       return lastStudent.getId();
+   }
+
+   private long findLastFacultyId() {
+       List<Faculty> faculties = facultyRepository.findAll();
+       Faculty lastFaculty = faculties.stream()
+               .max(Comparator.comparing(Faculty::getId))
+               .orElse(null);
+       if (lastFaculty == null) {
+           throw new NullPointerException("таблица с факультетами пуста");
+       }
+       return lastFaculty.getId();
+   }
+
+   @Test
+   void createStudentTest() {
+       Student student = new Student(findLastStudentId() + 1, "Имя", 15);
+       assertNotNull(this.testRestTemplate.postForObject(
+               "http://localhost:" + port + "/student", student, String.class));
+   }
+
+   @Test
+   void createFacultyTest() {
+       Faculty faculty = new Faculty(findLastFacultyId() + 1, "Факультет", "Черный");
+       assertNotNull(this.testRestTemplate.postForObject(
+               "http://localhost:" + port + "/faculty", faculty, String.class));
+   }
+
+   @Test
+   void getStudentInfoTest() {
+       assertNotNull(this.testRestTemplate.getForObject(
+               "http://localhost:" + port + "/student/1", String.class));
+   }
+
+
+   @Test
+   void getFacultyInfoTest() {
+       assertNotNull(this.testRestTemplate.getForObject(
+               "http://localhost:" + port + "/faculty/1", String.class));
+   }
+
+   @Test
+   public void editStudentTest() {
+    Student student = new  Student(findLastStudentId(), "Имя2", 20);
+       this.testRestTemplate.put("http://localhost:" + port + "/student", student);
+       Optional<Student> optionalStudent = studentRepository.findById(findLastStudentId());
```

Comment:
Проблема в том, что не передаёшь id в запросе
```suggestion
       long lastStudentId = findLastStudentId();
       Student student = new Student(lastStudentId, "Имя2", 20);
       this.testRestTemplate.put("http://localhost:" + port + "/student/" + lastStudentId, student);
       Optional<Student> optionalStudent = studentRepository.findById(lastStudentId);
```